fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreTest.java

```diff
@@ -3031,4 +3033,138 @@ public void metaDataVersionZero() throws Exception {
         }
     }
 
+    /**
+     * Test mode for {@link #clearOmitUnsplitRecordSuffix(ClearOmitUnsplitRecordSuffixMode)}.
+     */
+    public enum ClearOmitUnsplitRecordSuffixMode {
+        EMPTY, SAVE, DELETE
+    }
+
+    @EnumSource(ClearOmitUnsplitRecordSuffixMode.class)
+    @ParameterizedTest(name = "clearOmitUnsplitRecordSuffix [mode = {0}]")
+    public void clearOmitUnsplitRecordSuffix(ClearOmitUnsplitRecordSuffixMode mode) throws Exception {
```

Comment:
I'm not sure how these tests reproduce the error. In particular, it seems like the bug was only in the upgrade path that you wouldn't hit if the format version changed, and I think all of these test what happens to `omit_unsplit_record_suffix` only at the check version where the format version and the set of indexes are changed.

Reply:
Right. To be clear, this test _does not_ fail before the rest of the PR. It's just filling in a test for the basic upgrade mechanism to ensure against regression.

The other two tests _do_ fail before the `main` changes.

Reply:
Oh, I see. The `clearOmitUnsplitRecordSuffixTyped` test fails because the check here sets it to `true`: https://github.com/FoundationDB/fdb-record-layer/blob/982e20992e06a8a1daa673fa1e072f0ffd8c8a60/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java#L2532

Then in the same `checkVersion`, it sets it back to false, so it doesn't need to happen in, like, its own transaction.